### ISSUE TITLE
Toolbar icon design

### ISSUE THREAD

**Issue by CandyFace** [pencil2d/pencil#603]
Hi

Looking at issue #550 made me think that instead of fixing the icons, it would be better to give the toolbar a makeover. There are lots of decisions to be made when designing icons, so instead of me just bashing pixels without any consideration in what the rest of the pencil community thinks, I've decided to share my thoughts and progress.

I've made a bunch of **sketches** already, mainly to figure out the placement and how the icons should be illustrated. Nothing is finalized though, so try not to think too much about the style, color or shading. 

Rules i've come up with so far:
- Style should be consistent across all icons, if possible
- icons should be designed in size 24x24.
- Icons should be designed in a vector program or be convertible from raster to vector.

### Iterations:
![vector](https://cloud.githubusercontent.com/assets/1045397/22863977/3844f006-f149-11e6-895b-4048ba03dcc4.png)

These represent the look as you would see them in the program ie. in pixels
![pixel](https://cloud.githubusercontent.com/assets/1045397/22863978/386231a2-f149-11e6-9aa6-5ad0a261fefc.png)

100% scale:
![100%](https://cloud.githubusercontent.com/assets/1045397/22864003/7fe5417c-f149-11e6-80bf-49df843a510b.png)

I prefer the top right icon, as it is displayed in a way similar to the old icon but with more emphasis on tip.

### Mockup:
![toolbar](https://cloud.githubusercontent.com/assets/1045397/22863548/96c583d6-f142-11e6-82e8-3fa4c6cb46b1.png)

I'll keep updating the thread, showing more icons as I get them done. If no one disagree or comment, then I'll just go with the design I've mentioned already (the top right).

**Comment by Jose-Moreno** [comment]
@CandyFace due to this issue #597 I was also thinking of moving to a stage where we can begin changing Pencil2D's overall graphic design at least regarding it's icons. 

As you can read on the latest response by scribblemaniac: https://github.com/pencil2d/pencil/issues/597#issuecomment-279237490

he says there shouldn't be a problem with having SVG icons displayed. I honestly believe it would be for the best to have the icons as vector files to improve readability and help with HiDPI screen users.

Regarding the icons themselves I think you've taken a great first step. I agree with your rules as well, however considering SVG icons must also become a factor while designing them since the size limit shouldn't be an issue. 

It'd be fine if we had visualization templates for different resolutions and Operating Systems like Windows, OSX, Or common flavours of linux distros. If it's ok I'll post my own proposals to this thread so we can focus our efforts. Thanks a lot for your contribution!

**Comment by CandyFace** [comment]
You're welcome to post any kind of proposals as long as it it's within UI. 

Changing from png to svg would be fine by me too, actually it would only make it easier for me, as vector to pixel graphics typically requires post editing to fit specific resolutions. 

The icons will look good in any resolution, so scaling will not be a problem.

**Comment by CandyFace** [comment]
On a second thought, I think we should stick with raster graphics. SVG is indeed better for scaling but since it's just pure vector data, it will be slower than rendering raster graphics. Qt also mentions that scaling UI is a bad idea and should be avoided, it's better to stick with fixed resolutions. https://wiki.qt.io/Performance_tip_Images

I'll make sure to export icons in 3 different resolutions.

**Comment by scribblemaniac** [comment]
No, we should be using vector graphics. They will take fractionally longer to load initially, but I'm sure they are cached as bitmaps by Qt and thus won't really affect Pencil's performance except during startup (if at all). Scaling UIs are useful for UHD resolutions and for accessibility purposes. Maybe I'll do a benchmark if I have time. If, for some reason, SVGs are super slow, we can easily render them out beforehand, but it's much more difficult to go the other way.

P.S. The icons look awesome ❤️

**Comment by CandyFace** [comment]
Hmm I'm not sure whether sticking with png rather than svg will be more difficult to go about the scaling problem. 

I would assume that the scaling operation should only be executed once, in the initial setup when starting Pencil, and maybe later configureable in preferences. There should therefore be actual gain in using vector instead of raster.

Scaling UI will also still be possible if we stick with png, it just requires more setup for different resolutions rather than presumably scaling the icons dynamically.

If there truly is no performance penalty in switching to vector, and no weird or wonky scaling issues then sure we can do it.
Nevertheless, we should test it before moving further with that discussion.

What is your take on this @chchwy?

## Progress:
Eraser:
![eraser](https://cloud.githubusercontent.com/assets/1045397/22933182/082778b6-f2cb-11e6-8730-e0040a29c589.png)
![eraser 2x](https://cloud.githubusercontent.com/assets/1045397/22933184/084d0ff4-f2cb-11e6-8f72-dfe1b89b66cb.png)
![eraser 3x](https://cloud.githubusercontent.com/assets/1045397/22933183/0845e170-f2cb-11e6-9622-f5f34e7f1f87.png)

Pen:
![smaller](https://cloud.githubusercontent.com/assets/1045397/22933194/110ea65c-f2cb-11e6-83a7-a89b001d1175.png)
![pen 2x](https://cloud.githubusercontent.com/assets/1045397/22933195/112ffa78-f2cb-11e6-9c02-46381c139462.png)
![pen 3x](https://cloud.githubusercontent.com/assets/1045397/22933196/1136c70e-f2cb-11e6-847a-3048d771dfba.png)

**Comment by CandyFace** [comment]
Not a crazy amount of work today, was busy with study. I managed to make a new paint bucket though and also refined the color picker icon ;)

Color picker:
![eyedropper](https://cloud.githubusercontent.com/assets/1045397/22985613/3332f88a-f3a9-11e6-9f24-d17be0fc7826.png)
![color picker 2 2x](https://cloud.githubusercontent.com/assets/1045397/22985614/3355fe3e-f3a9-11e6-9133-ce02be1ac7a6.png)
![color picker 2 3x](https://cloud.githubusercontent.com/assets/1045397/22985615/335b6bf8-f3a9-11e6-8798-88af4d5fc8e4.png)

Paint bucket:
![bucket](https://cloud.githubusercontent.com/assets/1045397/22985626/387fc8f4-f3a9-11e6-8394-610257afd053.png)
![paint bucket 2x](https://cloud.githubusercontent.com/assets/1045397/22985625/387ebb58-f3a9-11e6-998f-8a17058ec14e.png)
![paint bucket 3x](https://cloud.githubusercontent.com/assets/1045397/22985627/38835320-f3a9-11e6-8b3a-93a7512f327c.png)

**Comment by Jose-Moreno** [comment]
@CandyFace These are great Oliver. Great work. Regarding the svg dilemma, I agree with performing the benchmark, however even if it doesn't work I think we should have a base SVG file with the icons, for editing purposes that anyone can access in the spirit of being open-source (further design choices should be filtered by us though). I will upload soon the logo as well since it should be available for everyone to use too.

However we should also discuss where these source files should be uploaded. I don't know if we should create a "graphic design" branch here on github or if anyone else has an idea to make it accessible as well as part of the source files, please speak up. :smiley:

**Comment by scribblemaniac** [comment]
Okay, benchmarks are done. The code for them is at https://github.com/scribblemaniac/pencil/tree/svgbenchmarks. A box plot of the results can be found at https://plot.ly/~scribblemaniac/7.embed. Looking at these results, it is apparent that the speed of rendering depends heavily on the complexity of the SVG file. For very simple icons, SVG rendering can be even faster than PNG rendering with scaling! However, the initial loading times for SVGs can get *relatively* high. Here's the timing of the two main ways to render SVGs:

QSvgRenderer: 0.13-19ms for initialization, 0.045-4ms to render
QIcon: 0.3-35ms for initialization, 0.001-0.025ms to render
PNGs: 0.002 for initialization and render

As you can see, the initialization time varies greatly between the simple SVG (1kB) and the complex SVG (680kB). I'm thinking that most of the icons we have will be much closer to the simple icon times, which are definitely acceptable. I would say as long as QIcon is able to load the SVG file in <5ms, then we can use QIcon with SVG files with no problems. If it takes longer than that, we may have to use a scaled PNG to start, and then load the SVG with QIcon after Pencil2d is finished loading.

TLDR; Benchmarks say SVG icons are fine 😋

**Comment by CandyFace** [comment]
That looks good @scribblemaniac, i'm not entirely sure about the size yet, but i'm estimating they'll be between 5-10kb, so that shouldn't be a problem. @Jose-Moreno, if we go with png, then i'll still upload svg for editing purposes, I do not intend to keep anything hostage ;)

## Progress: 
hand icon
![hand](https://cloud.githubusercontent.com/assets/1045397/23096422/7a5fe680-f61c-11e6-9d0e-498b68c0f1e3.png)
![hand 2x](https://cloud.githubusercontent.com/assets/1045397/23096424/7a8e1140-f61c-11e6-9134-3c5f226730b3.png)
![hand 3x](https://cloud.githubusercontent.com/assets/1045397/23096423/7a8c7344-f61c-11e6-97e5-03fa505095dd.png)

**Comment by CandyFace** [comment]
New smudge icon:
![smudge](https://cloud.githubusercontent.com/assets/1045397/23182607/76afb678-f879-11e6-9234-7c69aa89535d.png)
![smudge 2x](https://cloud.githubusercontent.com/assets/1045397/23182608/76dc34a0-f879-11e6-9a76-84e544a8900f.png)
![smudge 3x](https://cloud.githubusercontent.com/assets/1045397/23182609/76df0086-f879-11e6-80c4-b63033848373.png)

**Comment by Jose-Moreno** [comment]
@CandyFace I think every icon you've presented so far works fantastic except for those that involve realistic human parts. That's why usually there's this "cartoon glove" kind of thing on most software. 

By themselves they look like great illustrations, but as an icon I think it's better to lean towards simplicity without details like nails and avoiding more than two (2) gradients or smooth rendering of skin tones.

It is also perhaps a good rule of thumb to design those anatomical parts by using more curved lines or constructing those based on circular patterns.

Perhaps even using the old cartoon "4 finger" stylization could work better. Something like this but that fits the style you're going for:

![image](https://cloud.githubusercontent.com/assets/4912907/23267236/2104603a-f9b8-11e6-8b28-482d1f17b955.png)

**Comment by CandyFace** [comment]
@Jose-Moreno Thanks for the feedback! Yes you are absolutely correct, they don't work as icons. I was never happy with them tbh. it was just to show progress but you are right. A glove would be nice although white is quite tricky to use without outlines, so for consistency sake I would prefer to not have them. 

I'll try and see if I can make it work :)

**Comment by CandyFace** [comment]
Update:

Alright i've had some time to work on the icons again.. I've actually had a few of them done for a while now but... I'm just not sure I like them :P

anyway here's a few iterations of the hand icon. Shamelessly taken from @Jose-Moreno sketch because I like it :3

![hand skin](https://cloud.githubusercontent.com/assets/1045397/23555775/7d6b3fc4-002a-11e7-9720-9ae01f0de5b0.png)
![hand skin 2x](https://cloud.githubusercontent.com/assets/1045397/23555778/7d93b0a8-002a-11e7-8a43-7e8243852a8e.png)
![hand skin 3x](https://cloud.githubusercontent.com/assets/1045397/23555780/7d94fd14-002a-11e7-9f15-94e6546dbae8.png)

No clipping: 
![hand skin2](https://cloud.githubusercontent.com/assets/1045397/23555969/1f50d6dc-002b-11e7-90cf-51cd501c0cbf.png)
![hand skin2 2x](https://cloud.githubusercontent.com/assets/1045397/23555970/1f744022-002b-11e7-8757-a5151fb7134b.png)
![hand skin2 3x](https://cloud.githubusercontent.com/assets/1045397/23555971/1f74ffbc-002b-11e7-92e7-ae30bb2f1766.png)

Brush icon:
![brush](https://cloud.githubusercontent.com/assets/1045397/23555746/635c6306-002a-11e7-8540-6e697aced98d.png)
![brush 2x](https://cloud.githubusercontent.com/assets/1045397/23555748/638277e4-002a-11e7-910d-dce67f721ca9.png)
![brush 3x](https://cloud.githubusercontent.com/assets/1045397/23555747/6380785e-002a-11e7-8c57-ab1af4f89a0a.png)

I'm going to redo the brush though... I just don't like it :/

**Comment by CandyFace** [comment]
Last icons done
arrow tool:
![arrow](https://cloud.githubusercontent.com/assets/1045397/24800503/b2f3bb62-1ba0-11e7-81c8-d0c183d396ae.png)
![arrow 2x](https://cloud.githubusercontent.com/assets/1045397/24800504/b314f8cc-1ba0-11e7-9f40-e645674788cc.png)
![arrow 3x](https://cloud.githubusercontent.com/assets/1045397/24800505/b3168f3e-1ba0-11e7-97bd-25eca6e02417.png)

selection tool:


brush tool:
![brush](https://cloud.githubusercontent.com/assets/1045397/24800520/bc3ae948-1ba0-11e7-98e4-b02c99a2210f.png)
![brush 2x](https://cloud.githubusercontent.com/assets/1045397/24800521/bc3c003a-1ba0-11e7-9ef1-7053136e4d27.png)
![brush 3x](https://cloud.githubusercontent.com/assets/1045397/24800522/bc3fa442-1ba0-11e7-9456-9698ae2d9bab.png)

Clear tool:
![clear](https://cloud.githubusercontent.com/assets/1045397/24800556/d6a5f7d2-1ba0-11e7-8b07-9d7fe79a99bc.png)
![clear 2x](https://cloud.githubusercontent.com/assets/1045397/24800558/d6c8e4f4-1ba0-11e7-8fa7-b2e374b3c124.png)
![clear 3x](https://cloud.githubusercontent.com/assets/1045397/24800557/d6c6ba30-1ba0-11e7-8e5e-d2d85fd2b024.png)

Line tool:
![line tool](https://cloud.githubusercontent.com/assets/1045397/24800564/dc596524-1ba0-11e7-99e8-58f01b8ad715.png)
![line tool 2x](https://cloud.githubusercontent.com/assets/1045397/24800563/dc58b728-1ba0-11e7-9e5d-f2cf85953b37.png)
![line tool 3x](https://cloud.githubusercontent.com/assets/1045397/24800565/dc59e58a-1ba0-11e7-8a60-5ffdae9a65b6.png)

Hand tool:
![no outline](https://cloud.githubusercontent.com/assets/1045397/24800571/e33b7710-1ba0-11e7-9068-a36c17237cc1.png)
![no outline 2x](https://cloud.githubusercontent.com/assets/1045397/24800569/e33aa39e-1ba0-11e7-9923-bcd66f14bad8.png)
![no outline 3x](https://cloud.githubusercontent.com/assets/1045397/24800570/e33b4e7a-1ba0-11e7-96f6-6fed8b0e12e6.png)

Selection tool:
![selection](https://cloud.githubusercontent.com/assets/1045397/24800581/ebcebf4a-1ba0-11e7-8d79-a31b82369170.png)
![selection 2x](https://cloud.githubusercontent.com/assets/1045397/24800582/ebcfa91e-1ba0-11e7-8d29-f9c5a3f9b397.png)
![selection 3x](https://cloud.githubusercontent.com/assets/1045397/24800583/ebd26910-1ba0-11e7-9c19-a1b65a97687e.png)

Smudge tool:
![smudge 2](https://cloud.githubusercontent.com/assets/1045397/24800593/f3abfa66-1ba0-11e7-846e-7bfe9ce87e95.png)
![smudge 2 2x](https://cloud.githubusercontent.com/assets/1045397/24800592/f3aae9c8-1ba0-11e7-876f-f57cca3042e7.png)
![smudge 2 3x](https://cloud.githubusercontent.com/assets/1045397/24800594/f3ae9e38-1ba0-11e7-8bdf-73840dc96240.png)

Now I just need to clean some things so they all looks the same in Vector, then I'll make a PR which has both png and svg. I think i'll make a new folder in resources so it's not mixed up with the rest, then we can always clean it up later.

**Comment by J5lx** [comment]
I think the clear tool looks like some kind of rock. The others are good though. I wonder what symbol other programs use for ‘clear’.

**Comment by CandyFace** [comment]
As far as most drawing applications i've used, they don't have a clear tool, it's usually an action which hidden in the menu. I do agree though that it can resemble a rock. I can make it brighter, maybe that'll help.

**Comment by J5lx** [comment]
For reference, GIMP has this (in menu): 
![screenshot from 2017-04-07 15-05-01](https://cloud.githubusercontent.com/assets/2063777/24801268/c3c12440-1ba3-11e7-9d82-148effa4666f.png)
But it’s true that most software does not seem to have a symbol for that. Blender at least has a × to clear input fields but of course canvas ≠ input field.

**Comment by CandyFace** [comment]
So what about a trashcan?
![clear](https://cloud.githubusercontent.com/assets/1045397/24802906/d71b7b5c-1ba9-11e7-907b-8d84de61c701.png)
![clear 2x](https://cloud.githubusercontent.com/assets/1045397/24802907/d736e400-1ba9-11e7-8166-06b836450221.png)
![clear 3x](https://cloud.githubusercontent.com/assets/1045397/24802908/d737741a-1ba9-11e7-87a2-eefcb9f3235c.png)

It's simple and we've all seen one, so it should be easy to figure out what it does.

**Comment by J5lx** [comment]
Yeah, I think it’s a good option. Maybe a little more texture would be good (like the three, uh, vertical lines in [this symbolic one](http://fontawesome.io/icon/trash-o/)), but I think even the current one is already sufficiently recognizable (apart from that backdrop issue you noted).

**Comment by Jose-Moreno** [comment]
@CandyFace Thanks a lot for your work. I think the trashcan works. The symbol that Jakob describes could work but I see it more as those stereotypical trash cans from a big U.S city resting on an alley from the 80's hehe. 

A recycle symbol could help solve the issue of identifying the element even if we dismissed the trash can completely. And we're being eco-friendly lol.

On the matter of the blue lines, I really like the clean and modern look they give and the silhouette of the icons is working really nice.  I still think that having too much blending / gradients on the icons is not ideal though. 
Is there a way we can see the icons with flat colours only? If you don't have much time I'd like to try it out, can you upload a working version of your file? For making them pop I think the classic "cartoon" outline could work mixed with the flat colored version I'm proposing.

Thanks again, let me know about the file whenever you have some time! :smile:

**Comment by J5lx** [comment]
@Jose-Moreno Well, stereotypical means that it will be recognized :) Also I’m not insisting on exactly that design, it was just one possible example (though possibly the most common one). I think a recycle symbol would be more ambiguous than an 80s US trash can :)

Besides that, I agree that the version in edit 2 works well!

On the matter of flat vs. gradients, maybe icon themes à la GIMP (Edit/Preferences/Theme) or FileZilla (Edit/Settings…/Interface/Themes) could be an option too, at some point, though I have absolutely no idea how easy or difficult it might be to implement.

**Comment by CandyFace** [comment]
Thanks guys!

@Jose-Moreno
Alright, i'll try to make the icons flat, i'm not so sure about a cartoon outline but i'll make two versions anyway, then you guys can judge. 

@J5lx yeah themes would be cool, but it's definitely a task for another time. The implementation shouldn't be that bad though, insert a new tab (or add to the general tab) in the preferences window, add a dropdown menu to choose between "stylished" or simple.

**Comment by CandyFace** [comment]
@Jose-Moreno Here are two flat versions, the second is outlined.

flat version:
![pen flat](https://cloud.githubusercontent.com/assets/1045397/24815926/4e17d110-1bd7-11e7-9ccd-3028bb6bcf25.png)
![pen flat 2x](https://cloud.githubusercontent.com/assets/1045397/24815930/4e3a857a-1bd7-11e7-9481-a3989f943f24.png)
![pen flat 3x](https://cloud.githubusercontent.com/assets/1045397/24815933/4e3bb30a-1bd7-11e7-8470-f4bfa04818a8.png)
![bucket flat](https://cloud.githubusercontent.com/assets/1045397/24815931/4e3abdd8-1bd7-11e7-9573-b406b7479c24.png)
![bucket flat 2x](https://cloud.githubusercontent.com/assets/1045397/24815928/4e394d18-1bd7-11e7-9927-10a5a381cfed.png)
![bucket flat 3x](https://cloud.githubusercontent.com/assets/1045397/24815929/4e39fbb4-1bd7-11e7-8e52-dacf1be00045.png)
![eraser flat](https://cloud.githubusercontent.com/assets/1045397/24815927/4e2e4e04-1bd7-11e7-86da-94e2eb5933c9.png)
![eraser flat 2x](https://cloud.githubusercontent.com/assets/1045397/24815934/4e4449c0-1bd7-11e7-98e9-b24014f88f63.png)
![eraser flat 3x](https://cloud.githubusercontent.com/assets/1045397/24815935/4e514fbc-1bd7-11e7-8eee-5258a87c77f8.png)
![eyedropper flat](https://cloud.githubusercontent.com/assets/1045397/24815936/4e52beb0-1bd7-11e7-8563-93f9cddd7548.png)
![eyedropper flat 2x](https://cloud.githubusercontent.com/assets/1045397/24815937/4e53e8d0-1bd7-11e7-84ff-943ccebfc0d6.png)
![eyedropper flat 3x](https://cloud.githubusercontent.com/assets/1045397/24815938/4e549924-1bd7-11e7-899d-2ea8fb308bdd.png)

![brush flat](https://cloud.githubusercontent.com/assets/1045397/24815939/4e5df5e6-1bd7-11e7-9237-3c96d01c0fb4.png)
![brush flat 2x](https://cloud.githubusercontent.com/assets/1045397/24815940/4e680e96-1bd7-11e7-850c-ae0a1e3ae6d5.png)
![brush flat 3x](https://cloud.githubusercontent.com/assets/1045397/24815941/4e6aa944-1bd7-11e7-97b5-91cb43d63754.png)
![clear flat](https://cloud.githubusercontent.com/assets/1045397/24815942/4e6cb216-1bd7-11e7-97ee-163ad0b05b8b.png)
![clear flat 2x](https://cloud.githubusercontent.com/assets/1045397/24815943/4e6e8d3e-1bd7-11e7-853d-2681856c03fb.png)
![clear flat 3x](https://cloud.githubusercontent.com/assets/1045397/24815944/4e70dd1e-1bd7-11e7-824c-c6a5a613c924.png)
![ersaer flat](https://cloud.githubusercontent.com/assets/1045397/24815945/4e7a0c36-1bd7-11e7-8fe1-1b38d0b5e728.png)
![ersaer flat 2x](https://cloud.githubusercontent.com/assets/1045397/24815948/4e856810-1bd7-11e7-86cc-3263a431a2ec.png)
![ersaer flat 3x](https://cloud.githubusercontent.com/assets/1045397/24815946/4e80ca94-1bd7-11e7-8c8c-cc51670f792a.png)
![hand flat](https://cloud.githubusercontent.com/assets/1045397/24815947/4e827f56-1bd7-11e7-8d11-6bc0d245c7fb.png)
![hand flat 2x](https://cloud.githubusercontent.com/assets/1045397/24815949/4e86fef0-1bd7-11e7-9f71-09d2667b6f61.png)
![hand flat 3x](https://cloud.githubusercontent.com/assets/1045397/24815950/4e8a7e90-1bd7-11e7-8d1e-8ab4aa27b859.png)
![smudge flat](https://cloud.githubusercontent.com/assets/1045397/24815951/4e922df2-1bd7-11e7-8b13-ba029ea5c704.png)
![smudge flat 2x](https://cloud.githubusercontent.com/assets/1045397/24815952/4e986b90-1bd7-11e7-9b7c-2d64c506ee97.png)
![smudge flat 3x](https://cloud.githubusercontent.com/assets/1045397/24815953/4e98a48e-1bd7-11e7-9355-0d72950ed49f.png)

flat outlined version:
![brush flat_outline](https://cloud.githubusercontent.com/assets/1045397/24817280/3fd7383e-1bdc-11e7-87e6-7e22422a7473.png)
![brush flat_outline 2x](https://cloud.githubusercontent.com/assets/1045397/24817282/3ffc18b6-1bdc-11e7-841d-2a63a77550bf.png)
![brush flat_outline 3x](https://cloud.githubusercontent.com/assets/1045397/24817281/3ff99d8e-1bdc-11e7-8401-5b22ae879fd9.png)

![bucket flat_outline](https://cloud.githubusercontent.com/assets/1045397/24817322/61cc9218-1bdc-11e7-98c5-c8b5a44f5aab.png)
![bucket flat_outline 2x](https://cloud.githubusercontent.com/assets/1045397/24817325/61edb22c-1bdc-11e7-8f74-e41455eb8d2c.png)
![bucket flat_outline 3x](https://cloud.githubusercontent.com/assets/1045397/24817326/61edf4d0-1bdc-11e7-8624-e12c1ba461fc.png)

![clear flat_outline](https://cloud.githubusercontent.com/assets/1045397/24817157/b3a1296a-1bdb-11e7-81ee-9b893259e745.png)
![clear flat_outline 2x](https://cloud.githubusercontent.com/assets/1045397/24817163/b3bbd846-1bdb-11e7-86e2-2fcfbe4e4d56.png)
![clear flat_outline 3x](https://cloud.githubusercontent.com/assets/1045397/24817164/b3c26eae-1bdb-11e7-9450-4e015b5607cc.png)
![eraser flat_outline](https://cloud.githubusercontent.com/assets/1045397/24817165/b3d39364-1bdb-11e7-8e22-19d7f854505d.png)
![eraser flat_outline 2x](https://cloud.githubusercontent.com/assets/1045397/24817166/b3d4fa24-1bdb-11e7-9b33-c728666e4839.png)
![eraser flat_outline 3x](https://cloud.githubusercontent.com/assets/1045397/24817167/b3d647a8-1bdb-11e7-817a-0d9bdaef692b.png)
![eyedropper flat_outline](https://cloud.githubusercontent.com/assets/1045397/24817168/b3d79ba8-1bdb-11e7-81ec-120829ac712b.png)
![eyedropper flat_outline 2x](https://cloud.githubusercontent.com/assets/1045397/24817169/b3d94df4-1bdb-11e7-8b43-55699e0ccf04.png)
![eyedropper flat_outline 3x](https://cloud.githubusercontent.com/assets/1045397/24817170/b3dab9f0-1bdb-11e7-9d9f-a29254cd9eb4.png)
![hand flat_outline](https://cloud.githubusercontent.com/assets/1045397/24817300/4f23840a-1bdc-11e7-82ef-5be6de7a3913.png)
![hand flat_outline 2x](https://cloud.githubusercontent.com/assets/1045397/24817302/4f2630ce-1bdc-11e7-8b5d-b62853729bf5.png)
![hand flat_outline 3x](https://cloud.githubusercontent.com/assets/1045397/24817301/4f258f52-1bdc-11e7-87bb-2fa257c8df84.png)
![pen flat_outline](https://cloud.githubusercontent.com/assets/1045397/24817174/b3f38304-1bdb-11e7-803f-c7a062d248d5.png)
![pen flat_outline 2x](https://cloud.githubusercontent.com/assets/1045397/24817175/b3f405b8-1bdb-11e7-8106-55ca298e4382.png)
![pen flat_outline 3x](https://cloud.githubusercontent.com/assets/1045397/24817176/b3f5adb4-1bdb-11e7-9332-d6d6d9683c1f.png)
![smudge flat_outline](https://cloud.githubusercontent.com/assets/1045397/24817347/741d6082-1bdc-11e7-863e-9db4208c9b9c.png)
![smudge flat_outline 2x](https://cloud.githubusercontent.com/assets/1045397/24817349/743e0bca-1bdc-11e7-85e6-0db84efa6318.png)
![smudge flat_outline 3x](https://cloud.githubusercontent.com/assets/1045397/24817348/743d09b4-1bdc-11e7-882c-868bdd2bf23b.png)

**Comment by Jose-Moreno** [comment]
@CandyFace I liked how it the outline makes it pop. However I think if we were to test that route further there should still be "flat shadows" as well to help enhance the volume, For example on the eraser the bottom planes could be a slightly darker color to give that sense and so on. 

Looking at them, I continue to believe you nailed the shapes, now it's just a matter of providing enough detail for it to be understood at an icon level without making it too detailed. At this stage I'd leave the previous set you showed today while softening the shadows on the Pen and the brush, while we test them and gather more feedback while considering to improve their readability.

The only one that confuses me off a bit is the smudge icon. I think the idea is fine, but the index finger should stick out a lot more, since traditionally that's the most used finger to smudge a drawing, or rather the other two fingers should be pressed closer to the palm. Plus having a bolder indication of a thumb wouldn't hurt.

Regarding the trash can I guess Jakob's suggestion might come in handy, perhaps a trash can lid or similar could help to sell the idea of the trash can just to enhance it's silhouette. Right now it sort of resembles a glass.

Thank you for all your contributions. Really liking all of it!

**Comment by J5lx** [comment]
I think a grid-like structure might also work for the waste bin, somewhat like [here](http://i.stack.imgur.com/BVG7e.png). Basically just any kind of texture, as I mentioned before. (That’s not saying something else can’t do the job, though.)

**Comment by CandyFace** [comment]
About the trashcan/bin icon, I think you're putting too much thought into it. Logically there's no reason why a glass would be there. The icon resembles the bin you've seen on windows and mac. As for the lid idea, that would be more for a trashcan with garbage and stuff, this is supposed to resemble those you have in an office for paper and such and they usually don't have a lid, at least not here :)

it's probably my fault though, when I say trashcan, I actually mean an office bin. I could add a recycle icon though, if that makes it better but i don't think it's necessary.

@Jose-Moreno Okay i'll look into it.
About the smudge icon, i'll make the index finger longer. The thumb is hidden beneath, i'll made a shadow to indicate that.

**Comment by CandyFace** [comment]
@Jose-Moreno 
Okay here's a slightly more shaded version, I wouldn't call it flat though but it's definitely simplified.
I've also added some inward curved bevels or (three vertical lines :P) to the bin icon.   
![brush flat_outline](https://cloud.githubusercontent.com/assets/1045397/24829245/aaf8ee7c-1c6e-11e7-8a6f-898b483992fd.png)
![brush flat_outline 2x](https://cloud.githubusercontent.com/assets/1045397/24829249/ab18a6a4-1c6e-11e7-8b78-57a14e7d0c70.png)
![brush flat_outline 3x](https://cloud.githubusercontent.com/assets/1045397/24829247/ab18049c-1c6e-11e7-98b2-64c834c2760e.png)
![bucket flat_outline](https://cloud.githubusercontent.com/assets/1045397/24829251/ab1c9f2a-1c6e-11e7-8296-faaf3367e66a.png)
![bucket flat_outline 2x](https://cloud.githubusercontent.com/assets/1045397/24829250/ab191738-1c6e-11e7-8586-b4e40e04e154.png)
![bucket flat_outline 3x](https://cloud.githubusercontent.com/assets/1045397/24829248/ab1882b4-1c6e-11e7-9fa5-b5bc242a0428.png)
![clear flat_outline](https://cloud.githubusercontent.com/assets/1045397/24829318/67800f44-1c6f-11e7-9fef-d624e0197d38.png)
![clear flat_outline 2x](https://cloud.githubusercontent.com/assets/1045397/24829319/67a2a93c-1c6f-11e7-8022-3f6cf5745a10.png)
![clear flat_outline 3x](https://cloud.githubusercontent.com/assets/1045397/24829320/67a2cf5c-1c6f-11e7-9fbb-90599d599307.png)

![eraser flat_outline](https://cloud.githubusercontent.com/assets/1045397/24829254/ab307298-1c6e-11e7-8ce4-34750d60c51d.png)
![eraser flat_outline 2x](https://cloud.githubusercontent.com/assets/1045397/24829255/ab31d002-1c6e-11e7-85f0-5849da5526e9.png)
![eraser flat_outline 3x](https://cloud.githubusercontent.com/assets/1045397/24829256/ab323e2a-1c6e-11e7-9988-3df8840ec63a.png)
![eyedropper flat_outline](https://cloud.githubusercontent.com/assets/1045397/24829337/ab30bce8-1c6f-11e7-9c5b-75567af3c6f6.png)
![eyedropper flat_outline 2x](https://cloud.githubusercontent.com/assets/1045397/24829339/ab5370d0-1c6f-11e7-9685-efc16d0a8bed.png)
![eyedropper flat_outline 3x](https://cloud.githubusercontent.com/assets/1045397/24829338/ab520678-1c6f-11e7-8385-b74ed753dc01.png)

![hand flat_outline](https://cloud.githubusercontent.com/assets/1045397/24829260/ab4a779c-1c6e-11e7-85cc-417870ae7ba6.png)
![hand flat_outline 2x](https://cloud.githubusercontent.com/assets/1045397/24829261/ab4aeee8-1c6e-11e7-8192-de8a54982fd2.png)
![hand flat_outline 3x](https://cloud.githubusercontent.com/assets/1045397/24829262/ab4cd460-1c6e-11e7-8275-2996d01fd2e2.png)
![pen flat_outline](https://cloud.githubusercontent.com/assets/1045397/24829263/ab4eabf0-1c6e-11e7-9b9d-4b6a14a72a7d.png)
![pen flat_outline 2x](https://cloud.githubusercontent.com/assets/1045397/24829264/ab54acb2-1c6e-11e7-817d-f0844b025b66.png)
![pen flat_outline 3x](https://cloud.githubusercontent.com/assets/1045397/24829266/ab6223f6-1c6e-11e7-80e4-38c1d4136b4d.png)
![smudge flat_outline](https://cloud.githubusercontent.com/assets/1045397/24829265/ab60af80-1c6e-11e7-95ea-1065ccaf5fce.png)
![smudge flat_outline 2x](https://cloud.githubusercontent.com/assets/1045397/24829267/ab62a632-1c6e-11e7-804b-9214284e66f8.png)
![smudge flat_outline 3x](https://cloud.githubusercontent.com/assets/1045397/24829268/ab643da8-1c6e-11e7-83cc-d77f2e71d7b8.png)

**Comment by CandyFace** [comment]
+ More tweaking, 
+ new trash icon, now with recycle logo instead of bevelling.  
+ colored outline, not just brown anymore.

All icons has been cleaned up and should be ready now. I'll do another check tomorrow, just to see whether my eyes are not deceiving me, if they're not, then i'll probably upload svg + png tomorrow or one of the upcoming days. 

![screen shot 2017-04-09 at 22 11 27](https://cloud.githubusercontent.com/assets/1045397/24840694/e0d190b8-1d71-11e7-8dfe-5fcd534f8ff1.png)

**Comment by scribblemaniac** [comment]
@CandyFace What's the status of this? I would like to see these make it into the next release.

**Comment by CandyFace** [comment]
@scribblemaniac Hmm well since it was discussed in another thread that we should be focusing on bug fixing, I've put it on hold for now. All icons are more or less done though, just a bit more polishing

**Comment by CandyFace** [comment]
Alright i've made a PR now ;)
